In [2]:
import pandas as pd
import numpy as np
import seaborn as sns

In [14]:
# 加载数据集
data_train = pd.read_csv('train.csv')
data_test = pd.read_csv('test.csv')
# 将数据集合并
df = data_train.append(data_test)

In [8]:
# 检查数据格式
data_train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

### 数据解释
* __Passengerld__:乘客ID
* __Survived__:乘客是否生还
* __Pclass__:乘客登机(1/2/3等舱位)
* __Name__:乘客姓名
* __Sex__:乘客性别
* __Age__:年龄
* __SlbSp__:堂兄弟/妹个数
* __Parch__:父母与小孩的个数
* __Ticket__:票价
* __Cabin__:客舱
* __Embarked__:登船港口

In [10]:
# 查看数据常用的统计量信息
data_train.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

### 异常值初始观察(主要观察一下最大与最小值)
* __Fare__:船票的平均价格32.2,中位数14,平均值比中位数大很多,说明该特征分部是严重的右偏
* __Age__:最小值为0.17,最大值为80,0.17应该是不到一岁的婴儿,80则为老人家,这些极端数据需要进一步排查
* __Slbsp__:与Parch:Silbsp最大值为8,可能是异常,但又看Parch最大值为9.这两个特征同时出现,需要进一步排查

In [16]:
# 输出df数据集的观测值
'''
1.该数据集是一个DataFrame实例
2.数据的行索引是0到1308的一组数字
3.数据集中总共有1309行观测值
4.数据集中总共有12列变量,其中Age,Cabin,Embarked,Fare,Survived,Ticket有数据缺失
5.dtypess:给出了变量数据类型的汇总情况,可以用dtype方法获取每个变量的数据类型
6.memory usage:保存数据所耗费的类型
'''
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1309 entries, 0 to 417
Data columns (total 12 columns):
Age            1046 non-null float64
Cabin          295 non-null object
Embarked       1307 non-null object
Fare           1308 non-null float64
Name           1309 non-null object
Parch          1309 non-null int64
PassengerId    1309 non-null int64
Pclass         1309 non-null int64
Sex            1309 non-null object
SibSp          1309 non-null int64
Survived       891 non-null float64
Ticket         1309 non-null object
dtypes: float64(3), int64(4), object(5)
memory usage: 132.9+ KB


In [19]:
# 查看数据集的行列大小
f'合并后一共有{str(df.shape[0])}条数据'

'合并后一共有1309条数据'

In [18]:
# 统计各行缺失的数据量
pd.isnull(df).sum()

Age             263
Cabin          1014
Embarked          2
Fare              1
Name              0
Parch             0
PassengerId       0
Pclass            0
Sex               0
SibSp             0
Survived        418
Ticket            0
dtype: int64